packages/autocomplete-preset-algolia/src/types/algoliasearch.ts

```diff
@@ -0,0 +1,71 @@
+import * as ClientSearch from '@algolia/client-search';
+import type * as AlgoliaSearch from 'algoliasearch/lite';
+
+// turns any to unknown, so it can be used as a conditional
+type AnyToUnknown<TSubject> = (any extends TSubject ? true : false) extends true
+  ? unknown
+  : TSubject;
+
+type SearchClientShape = {
+  search: unknown;
+};
+
+type ClientLiteV5 = AnyToUnknown<
+  // @ts-ignore
+  ReturnType<typeof AlgoliaSearch.liteClient>
+>;
+type ClientSearchV5 = AnyToUnknown<
+  // @ts-ignore
+  ReturnType<typeof ClientSearch.searchClient>
+>;
+type ClientV5 = ClientLiteV5 extends SearchClientShape
+  ? ClientLiteV5
+  : ClientSearchV5 extends SearchClientShape
+  ? ClientSearchV5
+  : unknown;
+
+type PickForClient<
+  TMapping extends { v4: unknown; v5: unknown }
+> = ClientV5 extends SearchClientShape ? TMapping['v5'] : TMapping['v4'];
+
+export type SearchClient = PickForClient<{
+  // @ts-ignore
+  v4: AlgoliaSearch.SearchClient;
+  // @ts-ignore
+  v5: ClientV5;
+}>;
+
+export type MultipleQueriesQuery = PickForClient<{
+  // @ts-ignore
+  v4: ClientSearch.MultipleQueriesQuery;
+  // @ts-ignore
+  v5: AlgoliaSearch.LegacySearchMethodProps[number];
+}>;
+
+export type SearchForFacetValuesResponse = PickForClient<{
+  // @ts-ignore
+  v4: ClientSearch.SearchForFacetValuesResponse;
+  // @ts-ignore
+  v5: AlgoliaSearch.SearchForFacetValuesResponse;
+}>;
+
+export type SearchResponse<THit> = PickForClient<{
+  // @ts-ignore
+  v4: ClientSearch.SearchResponse<THit>;
+  // @ts-ignore
+  v5: AlgoliaSearch.SearchResponse; // should be generic, but isn't yet in the client
```

Comment:
apparently they're currently working on https://github.com/algolia/api-clients-automation/issues/853, maybe we avoid releasing incomplete compatibility? not sure if it matters much though, the type needs to be updated in the helper still after that's merged

Reply:
It seems this is now the case: https://github.com/algolia/algoliasearch-client-javascript/blob/next/packages/client-search/model/searchResponse.ts.